File: app/src/main/java/com/commit451/gitlab/viewHolder/GroupViewHolder.java
package com.commit451.gitlab.viewHolder;

import android.net.Uri;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.commit451.gitlab.App;
import com.commit451.gitlab.R;
import com.commit451.gitlab.model.api.Group;

import butterknife.BindView;
import butterknife.ButterKnife;

/**
 * View associated with a group
 */
public class GroupViewHolder extends RecyclerView.ViewHolder {

    public static GroupViewHolder inflate(ViewGroup parent) {
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.item_user, parent, false);
        return new GroupViewHolder(view);
    }

    @BindView(R.id.image)
    public ImageView image;
    @BindView(R.id.name)
    public TextView textName;

    public GroupViewHolder(View view) {
        super(view);
        ButterKnife.bind(this, view);
    }

    public void bind(Group group) {
        textName.setText(group.getName());

        if (group.getAvatarUrl() != null && !group.getAvatarUrl().equals(Uri.EMPTY)) {
            App.get().getPicasso()
                    .load(group.getAvatarUrl())
                    .into(image);
        }
    }
}

File: app/src/main/java/com/commit451/gitlab/viewHolder/GroupViewHolder.kt
package com.commit451.gitlab.viewHolder

import android.graphics.Color
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.annotation.ColorInt
import androidx.recyclerview.widget.RecyclerView
import coil.api.load
import com.commit451.addendum.recyclerview.bindView
import com.commit451.gitlab.R
import com.commit451.gitlab.model.api.Group
import com.commit451.gitlab.model.api.VISIBILITY_PUBLIC
import com.github.ivbaranov.mli.MaterialLetterIcon

/**
 * View associated with a group
 */
class GroupViewHolder(view: View) : RecyclerView.ViewHolder(view) {

    companion object {

        fun inflate(parent: ViewGroup): GroupViewHolder {
            val view = LayoutInflater.from(parent.context)
                    .inflate(R.layout.item_group, parent, false)
            return GroupViewHolder(view)
        }
    }

    val image: ImageView by bindView(R.id.image)
    private val iconLetter: MaterialLetterIcon by bindView(R.id.letter)
    private val textName: TextView by bindView(R.id.name)

    fun bind(group: Group, @ColorInt color: Int) {
        textName.text = group.name

        if (group.avatarUrl.isNullOrBlank() || group.visibility != VISIBILITY_PUBLIC) {
            image.visibility = View.GONE
            iconLetter.visibility = View.VISIBLE
            iconLetter.letter = group.name!!.substring(0, 1)
            iconLetter.letterColor = Color.WHITE
            iconLetter.shapeColor = color
        } else {
            iconLetter.visibility = View.GONE
            image.visibility = View.VISIBLE
            image.load(group.avatarUrl)
        }
    }
}
